src/states/Game.gd

```diff
@@ -62,20 +63,56 @@ func _on_player_move(grid_coords: Vector2):
 	level.handle_player_move(grid_coords)
 
 
+func hide_ui():
+	$CanvasLayer/UI.hide()
+	for element in get_tree().get_nodes_in_group("UI Elements"):
+		element.hide()
+
+
+func show_ui():
+	$CanvasLayer/UI.show()
+	for element in get_tree().get_nodes_in_group("UI Elements"):
+		element.show()
+
+
 func _on_LevelEnd_exit_reached_success(next_level_path: String):
 	get_tree().paused = true
-	$LevelComplete.update(next_level_path)
-	$LevelComplete/ColorRect.show()
-	$UI/Textbox.hide()
+	$CanvasLayer/LevelComplete.update_text(next_level_path)
+	$CanvasLayer/LevelComplete.show()
+	hide_ui()
+	Autosplitter.send_data("split")
+
+
+func _on_OptionsMenu_options_exited() -> void:
+	level.get_node("Player").update_animation_speed()
+	if options_return_to_game:
+		# We came from the game scene, so un-pause the game.
+		get_tree().paused = false
+		show_ui()
+	else:
+		$CanvasLayer/PauseMenu.show()
+		$CanvasLayer/PauseMenu/C/V/Buttons/OptionsButton.grab_focus()
+
+
+func _on_MenuButton_pressed() -> void:
+	# Go back to the main menu.
+	var error := get_tree().change_scene("res://src/states/Menu.tscn")
+	assert(not error)
 
 
 func _on_OptionsButton_pressed() -> void:
-	$PauseMenu/ColorRect.hide()
-	$OptionsMenu/ColorRect.show()
-	$OptionsMenu/ColorRect/C/V/BackButton.grab_focus()
+	get_tree().paused = true
+	options_return_to_game = true
```

Comment:
Not in love with this variable name, but I can't think of a better one.

Reply:
At least it causes the game to not crash LUL

I agree it's not great, but I also don't think it's much of an issue